server/src/main/java/org/apache/druid/server/QueryResource.java

```diff
@@ -337,27 +338,27 @@ protected ObjectMapper serializeDataTimeAsLong(ObjectMapper mapper)
     return mapper.copy().registerModule(new SimpleModule().addSerializer(DateTime.class, new DateTimeSerializer()));
   }
 
-  protected ResponseContext createContext(String requestType, boolean pretty)
+  protected QueryResponseContext createContext(String requestType, boolean pretty)
   {
     boolean isSmile = SmileMediaTypes.APPLICATION_JACKSON_SMILE.equals(requestType) ||
                       APPLICATION_SMILE.equals(requestType);
     String contentType = isSmile ? SmileMediaTypes.APPLICATION_JACKSON_SMILE : MediaType.APPLICATION_JSON;
-    return new ResponseContext(
+    return new QueryResponseContext(
         contentType,
         isSmile ? smileMapper : jsonMapper,
         isSmile ? serializeDateTimeAsLongSmileMapper : serializeDateTimeAsLongJsonMapper,
         pretty
     );
   }
 
-  protected static class ResponseContext
+  protected static class QueryResponseContext
```

Comment:
Could you please find this class a name which doesn't collide with `ResponseContext`, or, if there is no such name, cross-reference these classes in their Javadoc comments and explain the difference.

Reply:
Renamed to ResourceIOReaderWriter